### PR TITLE
[System.Reflection] CoreFX import for CustomAttributeTypedArgument type.

### PR DESCRIPTION
**Pull request by MaximLipnin**
Part of https://github.com/mono/mono/issues/9660

The changes:
- included corert version of CustomAttributeTypedArgument type;
- included related xunit tests from corefx System.Runtime;
- updated corert/Type.cs to have some required properties from corert Type.Internal.cs;
- removed obsolete System.Reflection/CustomAttributeTypedArgument.cs;
- moved back "array case" handling in CustomAttributeTypedArgument ctor (via related PR);
- re-enabled "Arrays" test in CustomAttributeDataTest on mobile.

### CODE REVIEW

File: mcs/class/corlib/System.Reflection/CustomAttributeTypedArgument.cs [hunk 1]
```diff
@@ -1,161 +0,0 @@
...
-		CustomAttributeTypedArgument (Type argumentType, object value)
-		{
-			if (argumentType == null)
-				throw new ArgumentNullException ("argumentType");
-
-			this.argumentType = argumentType;
-			this.value = (value == null) ? null : CanonicalizeValue (value);
-
-			// MS seems to convert arrays into a ReadOnlyCollection
-			if (value is Array) {
```
Comment (MaximLipnin) [inline mcs/class/corlib/System.Reflection/CustomAttributeTypedArgument.cs]: @akoeplinger As for [your comment](https://github.com/mono/mono/pull/10370#discussion_r213978561) about handling an "array case" in my prev PR - seems that it's still alive but they moved it to [here](https://github.com/dotnet/corert/blob/635cf21aca11265ded9d78d216424bd609c052f5/src/System.Private.Reflection.Core/src/System/Reflection/Runtime/CustomAttributes/RuntimeCustomAttributeData.cs#L189). Do I need to bring back this part as mono-conditional?

Comment (akoeplinger) [reply mcs/class/corlib/System.Reflection/CustomAttributeTypedArgument.cs]: Yes, and it seems the test failures confirm this: https://jenkins.mono-project.com/job/test-mono-pull-request-i386-osx/14257/testReport/junit/(root)/CustomAttributeDataTest/Arrays/

Btw. while you're at it please also reenable the test on mobile:
https://github.com/mono/mono/blob/3e841bf08911cd746bcf3473744706998fcc9bb7/mcs/class/corlib/Test/System.Reflection/CustomAttributeDataTest.cs#L64-L66

The referenced bugzilla bug was marked fixed a few years ago: https://bugzilla.xamarin.com/show_bug.cgi?id=10263

**Comment by MaximLipnin** [comment]
@monojenkins build failed

**Comment by marek-safar** [comment]
@monojenkins commit csproj and apidiff

**Comment by MaximLipnin** [comment]
Thanks for review!

**Pull request merged**